projects/storefrontlib/cms-structure/seo/structured-data/json-ld-script.factory.ts

```diff
@@ -78,16 +75,15 @@ export class JsonLdScriptFactory {
   }
 
   /**
-   * Sanitizes the given json-ld schema by leveraging the angular HTML sanitizer.
+   * Secure the given json-ld schema by encoding html characters (aka escaping), eg: <script> becomes &lt;script&gt;
    *
    * The given schema is not trusted, as malicious code could be injected (XSS)
    * into the json-ld script.
    */
-  sanitize(schema: {}): string {
-    return JSON.stringify(schema, (_key, value) =>
-      typeof value === 'string'
-        ? this.sanitizer.sanitize(SecurityContext.HTML, value)
-        : value
-    );
+  escapeHtml(schema: {}): string {
+    const renderer: Renderer2 = this.rendererFactory.createRenderer(null, null);
```

Comment:
not sure creating a renderer to create a detached div element is the best solution in terms of memory and performance. Could be an issue when you use with high frequency like https://github.com/angular-translate/angular-translate/issues/1843

other solutions to improve i could think: 
- create htmlElement from this.winRef.document.createElement
- create an html encoder with regex (challenging to cover all use cases)
- destroy() the local renderer if not done after return

Reply:
good idea! i saw it done on LaunchRenderStrategy.